Reporting: Update docs

sakjur opened the pull request:
+ Recommending sending a test email after report is saved for verification.
+ Added details about the different scheduling options.
+ Documented the experimental API we encourage users to use.
+ Image scale factor.

Feel free to apply grammar and style fixes directly on the PR.

Out-of-scope for this particular PR:
- Update documentation to mention that reporting is an early version still under heavy development. (#24045)
- Add examples to make it clear what to expect of a report. (#24045)

sakjur committed:
Reporting: Update docs

sakjur committed:
Merge branch 'master' into reporting-7-0-docs

# Conflicts:
#	docs/sources/enterprise/reporting.md

sakjur committed:
Reporting: Shuffle around + add version

jongyllen commented on docs/sources/enterprise/reporting.md:
```diff
@@ -55,16 +90,44 @@ The last saved version of the report will be sent to selected emails. You can us
 ## Rendering configuration
 
-When Grafana generates a report, it will render each panel separately and then put them together in a PDF file. You can configure the per-panel rendering request timeout and the maximum number of concurrent calls to the rendering service. These options are available in the [configuration]({{< relref "../installation/configuration.md">}}) file.
+When generating reports by each panel renders separately before being collected in a PDF. The per panel rendering timeout and number of concurrently rendered panels can be configured.
```
```suggestion
When generating reports, each panel renders separately before being collected in a PDF. The per panel rendering timeout and number of concurrently rendered panels can be configured.
```

sakjur replied:
```suggestion
When generating reports each panel renders separately before being collected in a PDF. The per panel rendering timeout and number of concurrently rendered panels can be configured.
```

Do we need that comma?

jongyllen commented on docs/sources/enterprise/reporting.md:
```diff
@@ -39,10 +39,45 @@ Currently only Organization Admins can create reports.
 1. Select the layout option for generated report: **Portrait** or **Landscape**.  
 1. Enter scheduling information. Options vary depending on the frequency you select.
 1. **Save** the report.
+1. **Send test mail** after saving the report to verify that the report looks like you expect it to.
```
I would like to add that it also can be used to verify that emails actually can be sent. Not sure how to phrase it though

sakjur replied:
```suggestion
1. **Send test mail** after saving the report to verify that the report works like you expect it to.
```

mayhaps?

AgnesToulet replied:
```suggestion
1. **Send test mail** after saving the report to verify that the whole configuration is working as expected.
```
Maybe? I agree that otherwise it's a bit redundant with the previous step `Preview the report to make sure it appears as you expect. Update if necessary`
And I think it should be "send test email" but this is what in the UI. Should we update it?

sakjur committed:
Update docs/sources/enterprise/reporting.md

Co-Authored-By: Jon Gyllenswärd <jon.gyllensward@grafana.com>

jongyllen reviewed (approved):
Looks good. Maybe comma, maybe not. We can probably live with this

sakjur commented:
:+1: Yeah, I guess.

AgnesToulet commented on docs/sources/enterprise/reporting.md:
```diff
@@ -39,10 +39,45 @@ Currently only Organization Admins can create reports.
+
+Scheduled reports can be sent on a weekly, daily, or hourly basis. You may also disable scheduling for when you either want to pause a report or send it via the API.
+
+All scheduling is approximative and indicates when the reporting service will start rendering the dashboard.
```
```suggestion
All scheduling indicates when the reporting service will start rendering the dashboard. It can take a few minutes to render a dashboard with a lot of panels.
```
It feels weird to start by `All scheduling is approximative`. It's not very positive about the feature.

sakjur replied:
🤔 I kind of prefer my variant here. For two reasons. One is that it's important to note that the _start_ of rendering might not be on the dot, it may shift back and forth a little based on a number of reasons (mostly forwards). The other thing is that one of the things we said we wanted to improve was setting clear expectations for the feature, and I think it'll lead to less frustration when the documentation acknowledges tradeoffs straight away.

oddlittlebird replied:
I prefer @AgnesToulet 's suggested edit.

sakjur replied:
I'll merge that then 🤷 But I still don't understand why, so I _will_ most likely do the same mistake in my next PR. I've learnt nothing here, and my response was specifically spelling out my reasons for my choices to open up for learning (even if it's just "it just feels better").

sakjur replied:
Just talked with @xlson and explained my concerns with the difference and he pointed out that my concern about the approximative start position. He'll open an issue with more details and we'll fix it in the backend instead of documenting it.

AgnesToulet commented:
Looks good!

We also discussed here: https://github.com/grafana/grafana-enterprise/pull/472#discussion_r400021280 about timerange issue. I tried to write a paragraph to explain this but not sure it's very clear. And I don't know where we should put this paragraph.

Preview:
>### Dashboard data timerange
>
>The dashboard data timerange is displayed at the top of the report. If your dashboard is set to use local browser timezone, it will use the Grafana server timezone.
>
> > If your Grafana server and your Renderer Plugin server are not in the same timezone, it could lead to two different displays between the top timerange and the graphs timeranges. In this case, we advise you to set your dashboard timezone to UTC to have display consistency.

Markdown:
```
### Dashboard data timerange

The dashboard data timerange is displayed at the top of the report. If your dashboard is set to use local browser timezone, it will use the Grafana server timezone.

> If your Grafana server and your Renderer Plugin server are not in the same timezone, it could lead to two different displays between the top timerange and the graphs timeranges. In this case, we advise you to set your dashboard timezone to UTC to have display consistency.
```

Or should we just add a paragraph in the troubleshooting section like `Different timezones in report timeranges`?

sakjur commented:
I think that looks pretty good, I'll try to incorporate that and see where it makes the most sense?

oddlittlebird commented on docs/sources/enterprise/reporting.md:
```diff
@@ -39,10 +39,45 @@ Currently only Organization Admins can create reports.
+Daily reports are generated once per day and takes two arguments:
+
+* **Time -** Time of day in 24 hours format when the report should be sent.
+* **Time zone-** Time zone for the **Time** argument.
```
```suggestion
* **Time zone -** Time zone for the **Time** argument.
```

oddlittlebird commented on docs/sources/enterprise/reporting.md:
```diff
@@ -39,10 +39,45 @@ Currently only Organization Admins can create reports.
+
+#### Hourly
+
+Hourly reports are generated once per hour and takes two arguments:
```
I am confused by your use of "arguments" here. Are these options in the UI that you fill in or select? If so, then they are probably fields. 

This is end user documentation, not developer notes.

sakjur replied:
I feel as I cannot verbatim just replace “arguments” with “fields” here, but it makes perfect sense that arguments is something that isn’t used outside of the command-line/API terminology. I’ll dig through the docs a little and write up something hopefully more user-centric tomorrow.

oddlittlebird commented on docs/sources/enterprise/reporting.md:
```diff
@@ -39,10 +39,45 @@ Currently only Organization Admins can create reports.
+
+* **Day -** Weekday which the report should be sent on.
+* **Time -** Time of day in 24 hours format when the report should be sent.
+* **Time zone-** Time zone for the **Time** argument.
```
```suggestion
* **Time zone -** Time zone for the **Time** argument.
```

oddlittlebird commented on docs/sources/enterprise/reporting.md:
```diff
@@ -55,16 +90,44 @@ The last saved version of the report will be sent to selected emails. You can us
 
 {{< docs-imagebox img="/img/docs/enterprise/reports_send_test_mail.png" max-width="500px" class="docs-image--no-shadow" >}}
 
+## API
```
```suggestion
## Reports API
```

oddlittlebird replied:
Are there going to be more Enterprise-only APIs? If so, then we will need to properly document this and others in a separate subsection.

sakjur replied:
Yes, that’s all but certain. Do you think we should add that already or should we wait until we have more/stable documented APIs? (I somewhat prefer the latter, otoh maybe that would mess up URLs or something like that?)

I’ll update the title.

oddlittlebird replied:
The thing is, we will need to remember to split this API out into a separate document when we get another one. It would mess up any links to that heading as well.

sakjur replied:
Then I’ll split it up and link to the API doc from here 🙂

sakjur committed:
Update docs/sources/enterprise/reporting.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

sakjur committed:
Update docs/sources/enterprise/reporting.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

sakjur committed:
Update docs/sources/enterprise/reporting.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

sakjur committed:
Update docs/sources/enterprise/reporting.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

sakjur committed:
Reporting: Split out HTTP API docs

sakjur committed:
arguments->fields

sakjur committed:
Merge branch 'master' into reporting-7-0-docs

jongyllen reviewed (approved):
Nice work!

AgnesToulet commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+
+## Reporting API
+
+### Manually send report
```
```suggestion
### Send report
```
I think the "manually" part is included in the fact that it's an API endpoint

AgnesToulet commented on docs/sources/enterprise/reporting.md:
```diff
@@ -55,16 +90,27 @@ The last saved version of the report will be sent to selected emails. You can us
 
+## Send report via the API
+
+Sending reports programmatically is possible with the [manually send report]({{< relref "http_api.md#manually-send-report" >}}) endpoint in the [Enterprise HTTP APIs]({{< relref "http_api.md" >}}).
```
```suggestion
Sending reports programmatically is possible with the [send report]({{< relref "http_api.md#send-report" >}}) endpoint in the [Enterprise HTTP APIs]({{< relref "http_api.md" >}}).
```
If you approve my comment above

AgnesToulet commented:
> 
> 
> I think that looks pretty good, I'll try to incorporate that and see where it makes the most sense?

I think you forgot about that? Or you didn't find anywhere to put it?

sakjur committed:
Update docs/sources/enterprise/http_api.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

sakjur committed:
Update docs/sources/enterprise/reporting.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

sakjur commented:
Yes! I did 😊 I'll add that 👍

sakjur committed:
Update docs/sources/enterprise/http_api.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

sakjur committed:
Reporting: Add section on time ranges

AgnesToulet commented on docs/sources/enterprise/reporting.md:
```diff
@@ -39,10 +39,45 @@ Currently only Organization Admins can create reports.
+Daily reports are generated once per day. All fields are required.
+
+* **Time -** Time of day in 24 hours format when the report should be sent.
+* **Time zone -** Time zone for the **Time** argument.
```
```suggestion
* **Time zone -** Time zone for the **Time** field.
```

AgnesToulet commented on docs/sources/enterprise/reporting.md:
```diff
@@ -39,10 +39,45 @@ Currently only Organization Admins can create reports.
+
+* **Day -** Weekday which the report should be sent on.
+* **Time -** Time of day in 24 hours format when the report should be sent.
+* **Time zone -** Time zone for the **Time** argument.
```
```suggestion
* **Time zone -** Time zone for the **Time** field.
```

AgnesToulet commented on docs/sources/enterprise/reporting.md:
```diff
@@ -39,10 +39,45 @@ Currently only Organization Admins can create reports.
+
+> Only available in Grafana Enterprise v7.0+.
+
+Reports which are scheduled to never be sent have no arguments and will not be sent to the scheduler. They may be manually generated from the **Send test email** prompt or via the API.
```
```suggestion
Reports which are scheduled to never be sent have no parameter and will not be sent to the scheduler. They may be manually generated from the **Send test email** prompt or via the API.
```

AgnesToulet reviewed (commented):
Nice work! Last comments around arguments / fields and I think we're good to go :)

oddlittlebird commented:
@AgnesToulet , "time range" is two words.

oddlittlebird commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+weight = 1000
++++
+
+# Enterprise HTTP APIs
```
Add an intro here and to each h2. It can be short, but it should set the context so the reader knows what to expect in the document or section.

oddlittlebird commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+
+Status Codes:
+
+code | description
```
```suggestion
Code | Description
```

oddlittlebird commented on docs/sources/enterprise/http_api.md:
````diff
@@ -0,0 +1,72 @@
+{"message":"Report was sent"}
+```
+
+Status Codes:
````
```suggestion
### Status codes
```

oddlittlebird commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+emails | string | Comma-separated list of emails to which to send the report to. Overrides the emails from the report. Required if **useEmailsFromReport** is not present.
+useEmailsFromReport | boolean | Send the report to the emails specified in the report. Required if **emails** is not present.
+
+**Example response:**
```
```suggestion
### Example response
```

oddlittlebird commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+
+JSON Body Schema:
+
+field name | type | description
```
```suggestion
Field name | Type | Description
```

oddlittlebird replied:
Can you be more specific about "type"? Data type? Entry type?

oddlittlebird commented on docs/sources/enterprise/http_api.md:
````diff
@@ -0,0 +1,72 @@
+}
+```
+
+JSON Body Schema:
````
```suggestion
### JSON body schema
```

oddlittlebird commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+
+This API endpoint is experimental and may be deprecated in a future release. On deprecation, a migration strategy will be provided and the endpoint will remain functional until the next major release of Grafana.
+
+**Example request:**
```
```suggestion
### Example request
```

oddlittlebird commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+
+## Reporting API
+
+### Send report
```
```suggestion
### Send a report
```

oddlittlebird commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+To access the API you need to authenticate yourself. For more information about API authentication, refer to [Authentication API]({{< relref "../http_api/auth.md" >}}).
+
+## Reporting API
+
```
Need an intro here. Something like

This API allows you to interact programmatically with the [Reporting](LINK) feature.

oddlittlebird commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+
+Generate and send a report. This API waits for the report to be generated before returning. You are recommended to set the client's timeout to 60 seconds or more.
+
+This API endpoint is experimental and may be deprecated in a future release. On deprecation, a migration strategy will be provided and the endpoint will remain functional until the next major release of Grafana.
```
You need to lead with this. Is it just reports/email that is experimental, or is it the whole reporting API? Put it as high up in the hierarchy as makes sense.

sakjur replied:
Just the `POST /api/reports/email` endpoint.

oddlittlebird commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+
+`POST /api/reports/email`
+
+Generate and send a report. This API waits for the report to be generated before returning. You are recommended to set the client's timeout to 60 seconds or more.
```
```suggestion
Generate and send a report. This API waits for the report to be generated before returning. We recommend that you set the client's timeout to at least 60 seconds.
```

oddlittlebird commented on docs/sources/enterprise/http_api.md:
````diff
@@ -0,0 +1,72 @@
+```http
+POST /api/reports/email HTTP/1.1
+Accept: application/json
+Content-Type: application/json
````
```suggestion
Content-type: application/json
```

sakjur replied:
The name of the header is `Content-Type`, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type

oddlittlebird commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+
+field name | type | description
+---------- | ---- | -----------
+id | string | ID of the report to send (same as in the URL when editing a report, not to be confused with the ID of the dashboard). Required.
```
```suggestion
id | string | ID of the report to send. It is the same as in the URL when editing a report, not to be confused with the ID of the dashboard. Required.
```

oddlittlebird commented on docs/sources/enterprise/http_api.md:
```diff
@@ -0,0 +1,72 @@
+
+code | description
+---- | -----------
+200 | Report was sent
```
Put periods at the ends of these sentences.

oddlittlebird commented on docs/sources/enterprise/reporting.md:
```diff
@@ -55,18 +90,37 @@ The last saved version of the report will be sent to selected emails. You can us
 
+## Send report via the API
+
+Sending reports programmatically is possible with the [send report]({{< relref "http_api.md#send-report" >}}) endpoint in the [Enterprise HTTP APIs]({{< relref "http_api.md" >}}).
```
```suggestion
You can send reports programmaticallywith the [Reporting API]({{< relref "http_api.md#send-report" >}}) in the [Enterprise HTTP APIs]({{< relref "http_api.md" >}}).
```

sakjur replied:
I'd like to refer to the exact endpoint, not the API as a whole, made the change to active voice but kept the endpoint reference.

oddlittlebird reviewed (changes requested):
I was unclear about the API doc. 

Like in the HTTP APIs section, we will eventually need one topic per API. This one looks good, except that it should be in a documentation section called Enterprise APIs (in the sidebar) and preferably in its own folder as well with an _index.md file. This will give us a good place to store future Enterprise API docs as well.

The current H2 Reporting API should be switched to an H1, then move the paragraph about Authentication down to under the bits about what the Reporting API does and the experimental bit.

sakjur committed:
Update docs/sources/enterprise/reporting.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

sakjur committed:
Update docs/sources/enterprise/reporting.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

sakjur committed:
Update docs/sources/enterprise/reporting.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>

sakjur committed:
Update docs/sources/enterprise/http_api.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

sakjur committed:
Update docs/sources/enterprise/http_api.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

sakjur committed:
Update docs/sources/enterprise/http_api.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

sakjur committed:
Update docs/sources/enterprise/http_api.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

sakjur committed:
Minor

sakjur committed:
Move API ref to HTTP API dir

Noticed there is a precedence for putting enterprise APIs together with the other APIs that I previously missed. Adhering to that.

sakjur committed:
minor

sakjur commented:
I noticed there is a precedence for putting Enterprise APIs under the HTTP APIs, so I did that with this one as well. The section about enterprise APIs still needs an introduction, but that's out of scope for this PR.

I've tried to apply or respond to all change requests that has come in so far and feel pretty confident that this is in a state where it's good enough for us to merge. I'll wait until tomorrow to gather some final feedback before merging 🙂

oddlittlebird commented on docs/sources/enterprise/reporting.md:
```diff
@@ -55,18 +90,37 @@ The last saved version of the report will be sent to selected emails. You can us
+
+The page header of the report displays the time range for the dashboard's data queries. Dashboards set to use the browser's time zone will use the time zone on the Grafana server.
+
+If the time zone is set differently between your Grafana server and its remote image renderer, then the time ranges in the report may be different between the page header and the time axes in the panels. We advise always setting the time zone to UTC for dashboards when using a remote renderer to avoid this. 
```
```suggestion
If the time zone is set differently between your Grafana server and its remote image renderer, then the time ranges in the report might be different between the page header and the time axes in the panels. We advise always setting the time zone to UTC for dashboards when using a remote renderer to avoid this. 
```

oddlittlebird replied:
Use "might" instead of "may" when talking about possibilities. "May" has connotations of permission. "Yes, you may do that" vs. "That might happen."

oddlittlebird commented on docs/sources/http_api/reporting.md:
```diff
@@ -0,0 +1,69 @@
+
+> Reporting is only available in Grafana Enterprise. Read more about [Grafana Enterprise]({{< relref "../enterprise" >}}).
+
+The Reporting API allows for programmatic access to [Reporting]({{< relref "../enterprise/reporting.md" >}}) functionality.
```
Switch the order of these two first paragraphs.

oddlittlebird commented on docs/sources/http_api/reporting.md:
```diff
@@ -0,0 +1,69 @@
+
+The Reporting API allows for programmatic access to [Reporting]({{< relref "../enterprise/reporting.md" >}}) functionality.
+
+## Send report
```
```suggestion
## Send a report
```

sakjur replied:
Whoops, had forgotten to push that fix 😇

oddlittlebird reviewed (approved):
Looks great. I added a few minor edits.

sakjur committed:
Update docs/sources/enterprise/reporting.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

sakjur committed:
flip paragraph order for Reporting API introduction

AgnesToulet commented on docs/sources/enterprise/reporting.md:
```diff
@@ -55,18 +90,37 @@ The last saved version of the report will be sent to selected emails. You can us
 
+## Report time range
+
+Reports uses the saved time range of the dashboard. Changing the time range of the report is done by saving a modified time range to the dashboard.
```
```suggestion
Reports use the saved time range of the dashboard. Changing the time range of the report is done by saving a modified time range to the dashboard.
```

AgnesToulet reviewed (approved):
LGTM! I added one last catch.

sakjur committed:
Update docs/sources/enterprise/reporting.md

Co-authored-by: Agnès Toulet <35176601+AgnesToulet@users.noreply.github.com>
